### DISCUSSION TITLE
Adding relative path to microfe crud service

### DISCUSSION THREAD

**Discussion by rmuller-vu** [mia-platform/community#560] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.102.8

### Description

We would like to be able to use relative path for the bk-crud-client component. We are trying to do `./v2/version-approvals`.  We have a 2 different routes to access the microfe. 

![image](https://github.com/user-attachments/assets/362930b4-4ec7-4c1e-af24-2274c061e65e)

Internal: `https://internal-host/mfe-application/version-approval`
Public: `https://public-host/prefix/mfe-application/version-approval`

Our public facing route goes through an api manager so the prefix is needed. We are able to use relative pathing in all areas except the components. Button with type http and the bk-crud-client. Is there a way to use relative path for the components without having to make a custom component?

**Comment by epessina** [comment, marked as answer]
Hi @rmuller-vu , sorry for the late reply.

URLs in back-kit components are computed relative to the page's origin (i.e., `new URL(path, window.location.origin)`) so they cannot consider any prefix. 

Possible solutions that come to my mind are:
- duplicating the configuration to add the prefix
- try and leverage API Gateway to rewrite calls from `public-host/v2/version-approvals` to `public-host/prefix/v2/version-approvals`
- making a custom web component

Let me know if you need further clarification
